File: course-cpp/20-classes.cpp
#include <iostream>
using namespace std;

class Hello {
public:

    // constructor
    Hello() {
        cout << "I am a constructor" << endl;
    }

    // instance method
    void hi() {
        cout << "I am an instance method!" << endl;
    }

    static void hello() {
        cout << "I am a static method!\n";
    }
};

int main() {

    Hello::hello();

    // create an instance of the class
    Hello hello;

    // call the instance method hi
    hello.hi();
    return 0;
}

File: course-python/20-classes.py
class MySuperClass():

    def __init__(self):
        print("I'm a super constructor")

    def superMethod(self):
        print("I'm a super method");

class MyClass(MySuperClass):

    # self is required to instantiate
    def __init__(self):

        # calls constructor of the super class
        super().__init__()
        print("I'm a constructor")

    def method(self):
        print("I'm a method")

Instance = MyClass()
Instance.method()
Instance.superMethod()
